src/impress.js

```diff
@@ -583,6 +615,110 @@
             return goto( next, undefined, "next", origEvent );
         };
 
+        // Swipe for touch devices by @and3rson.
+        // Below we extend the api to control the animation between the currently
+        // active step and a presumed next/prev step. See touch plugin for
+        // an example of using this api.
+
+        // Helper function
+        var interpolate = function( a, b, k ) {
+            return a + ( b - a ) * k;
+        };
+
+        // Animate a swipe.
+        //
+        // Pct is a value between -1.0 and +1.0, designating the current length
+        // of the swipe.
+        //
+        // If pct is negative, swipe towards the next() step, if positive,
+        // towards the prev() step.
+        //
+        // Note that pre-stepleave plugins such as goto can mess with what is a
+        // next() and prev() step, so we need to trigger the pre-stepleave event
+        // here, even if a swipe doesn't guarantee that the transition will
+        // actually happen.
+        //
+        // Calling swipe(), with any value of pct, won't in itself cause a
+        // transition to happen, this is just to animate the swipe. Once the
+        // transition is committed - such as at a touchend event - caller is
+        // responsible for also calling prev()/next() as appropriate.
+        var swipe = function( pct ) {
```

Comment:
I think the name `swipe` may be a bit misleading. It took me a while to realise what this method is doing and why it needs to be in impress.js core (not in some external mobile plugin).

The name this function adds some 'swipe' (or touch) navigation. But it seems (as far as I understand) moves the transition between two steps (for example while user is swiping).

I feel a bit unsure about having it in main impress API, as it doesn't seem like it would be used by users, but rather then by plugins, but I don't really know how to approach it other way.

It it used by multiple plugins? If it's only for swipe maybe it would make sense to move it into mobile/swipe plugin that adds support for that?
I guess what would be needed for that in public API would be some  additional `move` method , that would allow impress.js to move into any given position (like `impress().move({x: 100, y: 200, z: 300, rotateX, scale...}`). But... that seems like other work I guess.

@henrikingo You probably have a bigger picture. Does it make sense to keep it in public API?

Reply:
It's only used by the swipe plugin, so renaming or changing arguments is feasible here.

You're right that at this point your idea of a impress().move() function would really be handy, but it seems to me like a big change! In any case, as you can see from the first half of the swipe() function, it's not just about animating a swipe, but it's also related to the step concept. What really happens is a prestepleave -> swipe -> stepleave (or not, if cancelled). So what is now the swipe plugin, will still need to interact with a higher level part of the code that is in charge of the (pre)stepleave events, and only that part can then animate the actual swipe. From that point of view swipe() truly is an equal sibling to prev() and next().

Reply:
As it seems that the matter is more complicated, I don't mind leaving it as it is.
But I would try to think of some way to keep such methods more 'private' - at least not being so exposed to users.
I guess the easiest way would be to not document them in official public API, only as part of API docs (if that makes sense)? Or - if there will be more methods like that keep them separate from rest of public API?